### PR TITLE
Btc

### PR DESCRIPTION
**Pull request by dimalvovs**
- update load10Xcoords to work with both v1 and v2 versions of spaceranger;
- update container definitions;
- update tests;
- place spatial vignettes to /inst

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,4 +9,17 @@ RUN sudo apt-get update -y && \
...
     apt-get install libhdf5-dev build-essential patch -y
 
-RUN Rscript -e 'devtools::install_deps(); devtools::install()'
+RUN Rscript -e 'devtools::install_deps()'
+
+#https://github.com/r-lib/devtools/issues/2395
+RUN Rscript -e 'devtools::install()'
+
+#optoional packages - for vizualization
+RUN Rscript -e 'install.packages("patchwork");\
```
Comment (atuldeshpande) [inline Dockerfile:18]: I was wondering, if our intent is to separate the computation and visualizations for the SpaceMarkers in separate scripts/markdowns, should we create two separate dockerfiles accordingly, one for only the computation (I think the existing one), and the other for visualizations (we could invoke the visualization dockerfile each time we want to plot something?)

File: R/preprocessing.R [hunk 2]
```diff
@@ -90,20 +91,40 @@ load10XExpr<- function(visiumDir=NULL,
...
     scale_values <- jsonlite::read_json(scale_json)
-    scale_dia <- scale_values$spot_diameter_fullres
     scale_factor <- scale_values[grepl(resolution, names(scale_values))][[1]]
-    coord_file <- dir(paste0(visiumDir,'/spatial'),
-                        pattern="tissue_positions_list.csv",full.names = TRUE)
-    coord_values <- read.csv(coord_file,header = FALSE)
+    coord_file <- dir(spatial_dir,
+                        pattern = tissue_pos_name, full.names = TRUE)
+
+    coord_values <- read.csv(coord_file, header = has_header)
```
Comment (atuldeshpande) [inline R/preprocessing.R:127]: Do we need additional error handling here if someone gives the wrong version number?

File: Dockerfile (hunk 1 above)
Comment (dimalvovs) [reply Dockerfile:18]: Do we want to be able to run spacemarkers compute+viz in one go (in one container)? This is 2 months old, unfortunately I don't recall why (if) our intent was was to split the relatively small program into two containers.

File: R/preprocessing.R (hunk 2 above)
Comment (dimalvovs) [reply R/preprocessing.R:127]: Sounds like a valid even if not critical improvement!

**Review by atuldeshpande — APPROVED** [review]

**Pull request merged**